[Services] Add SMSup  provider

nmargaris95 opened the pull request:
Created new sms service with smsUp provider. This includes sms test.

Commit:
Sms Service for new provider and new sms Test (smsUp)

Commit:
Sms Service for new provider and new sms Test (smsUp) (2)

Commit:
Error handling for new Sms Service (smsUp)

Commit:
Add "using" for string content (autopilot suggestions)

travlos commented on src/Indice.Services/SmsServiceSmsUP.cs:
```diff
@@ -0,0 +1,227 @@
+        if (string.IsNullOrWhiteSpace(destination))
+            throw new ArgumentNullException(nameof(destination), "Recipient is empty.");
+
+        if (!PhoneNumber.TryParse(destination, out var phone))
```
Validate ApiKey & To

travlos replied:
Δεν έχεις κάνει validate ότι ο Settings.Sender! δεν είναι κενός.

travlos commented on src/Indice.Services/SmsServiceSmsUP.cs:
```diff
@@ -0,0 +1,227 @@
+            var errorId = error?.ErrorId ?? response.ErrorId;
+            var errorMsg = error?.ErrorMsg ?? response.ErrorMsg;
+            Logger.LogWarning("SmsUp error: {ErrorId} - {ErrorMsg}", errorId, errorMsg);
+            string userFriendlyMessage = errorId switch {
```
Move this in the class SmsApiResultItem

travlos reviewed (changes requested):
Fix the suggested codeql findings as well the two comments

Commit:
Applied recommended PR fixes

travlos commented on src/Indice.Services/SmsServiceSmsUP.cs:
```diff
@@ -0,0 +1,264 @@
+        .ToArray();
+
+        List<SmsUpMessage> messages = new List<SmsUpMessage>();
+        foreach (var recipient in recipients) {
+            SmsUpMessage smsUpMessage = new SmsUpMessage {
+                From = sender?.Id ?? Settings.Sender!, // fallback sender
+                To = recipient,
+                Text = body ?? subject ?? string.Empty,
+                Custom = Guid.NewGuid().ToString()
+            };
+            messages.Add(smsUpMessage);
+        }
```
Κάνε το fix που σου προτείνει

travlos committed:
Refactor SmsServiceSmsUP.cs for improved clarity

Added a check for `Settings.Sender` to ensure it's not null or whitespace. Simplified the handling of the `recipients` variable by removing unnecessary null checks and directly splitting the `destination` string. Refactored the creation of the `messages` list to use LINQ, enhancing readability and conciseness.

travlos committed:
Merge branch 'develop' into feature/smsUP

travlos reviewed (approved):
It's fine
